### PR TITLE
Docs Restructuring

### PR DESCRIPTION
**Pull request by AshwinSriram11**
## Description

As described in #6628, I have made the necessary changes for restructuring in this repository as well as in the website repository. As a draft, I have only made the basic structure as described in the proposal. 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

### CODE REVIEW

**Comment by ArthurSens** [comment]
Nice start! It's a big change so the review will be made in several parts 😬 


> As described in #6628, I have made the necessary changes for restructuring in this repository as well as in the website repository. As a draft, I have only made the basic structure as described in the proposal.

I've noticed that you haven't added the beginner and advanced sections that were mentioned in your proposal. Or maybe you added but with different names? I'm a bit confused.

There are also a lot of changes that seem unrelated to re-organizing sections, e.g new line breaks, bulleted lists changing from `*` to `-`, italics and bold marks with similar changes, etc. Is this all necessary?

**Comment by AshwinSriram11** [comment]
> Nice start! It's a big change so the review will be made in several parts 😬
> 
> > As described in #6628, I have made the necessary changes for restructuring in this repository as well as in the website repository. As a draft, I have only made the basic structure as described in the proposal.
> 
> I've noticed that you haven't added the beginner and advanced sections that were mentioned in your proposal. Or maybe you added but with different names? I'm a bit confused.
> 
> There are also a lot of changes that seem unrelated to re-organising sections, e.g new line breaks, bulleted lists changing from `*` to `-`, italics and bold marks with similar changes, etc. Is this all necessary?

My bad! I think these line breaks, list changes,etc. were caused by some formatting extension locally. I will fix this with my next commit. Also, I will change the names of the section.

**Comment by AshwinSriram11** [comment]
> > Nice start! It's a big change so the review will be made in several parts 😬
> > > As described in #6628, I have made the necessary changes for restructuring in this repository as well as in the website repository. As a draft, I have only made the basic structure as described in the proposal.
> > 
> > 
> > I've noticed that you haven't added the beginner and advanced sections that were mentioned in your proposal. Or maybe you added but with different names? I'm a bit confused.
> > There are also a lot of changes that seem unrelated to re-organising sections, e.g new line breaks, bulleted lists changing from `*` to `-`, italics and bold marks with similar changes, etc. Is this all necessary?
> 
> My bad! I think these line breaks, list changes,etc. were caused by some formatting extension locally. I will fix this with my next commit. Also, I will change the names of the section.

To be honest, as I mentioned in the last comment to change the changes mentioned. However, I think that these changes are fine because although this does not affect the readability of the website as things are the same, neither does it affect the preview in Github. But if someone wanted to read the markdown file as it is, this provides a little better readability(use of proper whitespaces, line spaces,etc). I know that this was a mistake from my side but I think it would be better to include this. It doesn't make that big of a difference and you would need to read the files but I think this can be included.

File: LICENSE [hunk 1]
```diff
@@ -1,181 +1,193 @@
...
+toc: true
+title: License
+menu:
+  docs:
+    parent: community
+lead: ""
+images: []
+draft: false
+---
+
```
Comment (ArthurSens) [inline LICENSE:12]: I'm not sure if we're allowed to make changes to the LICENSE file 🤔

**Review by ArthurSens — COMMENTED** [review]
Agreed it doesn't change the content once the markdown is rendered, but the PR could have been a lot easier to review if it was only +50-60 than +400-340. We could make changes to the markdown content in a separate PR to keep the scope of each PR as minimal as possible, and therefore easier to review.

It might seem slow to open multiple PRs for things that seem related to each other, but small scopes = smooth reviews and smooth reviews = fast progress.

File: LICENSE (hunk 1 above)
Comment (AshwinSriram11) [reply LICENSE:12]: Without this, we won't be able to include it. Should I remove it?

**Comment by AshwinSriram11** [comment]
The latest commit has added Developer and Platform Guide according to the proposal. So, headers were mainly changes for this commit(for the purpose of ordering).

**Comment by AshwinSriram11** [comment]
This commit has changed Prologue into Getting-Started and fixed some links according to the change.

**Review by ArthurSens — APPROVED** [review]
As you mentioned in our call @AshwinSriram11, we need to fix the navigation header settings for API reference, but this can be fixed in a follow up PR.

Let's open an issue and add it to our project's milestone

**Review by nicolastakashi — APPROVED** [review]

**Pull request merged**